File: .Net/UnitTest/src/Clothe.Application/DeleteClothe/DeleteClotheCommandHandler.cs
using MediatR;
using Clothes.Application.Results;
using Clothes.Domain.Entities;
using Clothes.Domain.Repositories;

namespace Clothes.Application.DeleteClothe;

public class DeleteClotheCommandHandler(IRepository<ClotheEntity> _repository, DeleteClotheCommandValidator _validator) : IRequestHandler<DeleteClotheCommand, Result>
{
    public async Task<Result> Handle(DeleteClotheCommand request, CancellationToken cancellationToken)
    {
        var isValid = await _validator.ValidateAsync(request, cancellationToken);
        if (!isValid.IsValid)
        {
            return new Result(false, "Invalid data", isValid.Errors);
        }
        var Clothe = await _repository.GetByIdAsync(request.Id, cancellationToken);
        if (Clothe is null)
        {
            return new Result(false, "Clothe not found");
        }

        _repository.Delete(Clothe);
        await _repository.SaveChangesAsync(cancellationToken);
        return new Result(true, "Clothe deleted successfully", Clothe.Id);
    }
}

File: Java/ClotheSample/src/main/java/application/delete/DeleteClotheCommandHandler.java
package application.delete;

import application.create.CreateClotheCommand;
import application.results.Result;
import domain.entities.ClotheEntity;
import domain.repositories.IRepository;
import org.springframework.stereotype.Service;

import javax.validation.ConstraintViolation;
import javax.validation.Validator;
import java.util.Optional;
import java.util.Set;
import java.util.UUID;

@Service
public class DeleteClotheCommandHandler {

    private final IRepository repository;
    private final Validator validator;

    public DeleteClotheCommandHandler(IRepository repository, Validator validator) {
        this.repository = repository;
        this.validator = validator;
    }

    public Result handle(DeleteClotheCommand request) {
        // Validate the command
        Set<ConstraintViolation<DeleteClotheCommand>> violations = validator.validate(request);
        if (!violations.isEmpty()) {
            return new Result(false, "Invalid data", null);
        }

        // Fetch the entity from the repository
        Optional<ClotheEntity> clothe = repository.findById(request.getId());
        if (clothe.isEmpty()) {
            return new Result(false, "Clothe not found", null);
        }

        // Delete the entity
        repository.delete(clothe.get());

        return new Result(true, "Clothe deleted successfully", clothe.get().getId());
    }
}
